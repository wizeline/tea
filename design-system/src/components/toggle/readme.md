# wz-toggle



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description | Type      |
| ---------- | ---------- | ----------- | --------- |
| `checked`  | `checked`  |             | `boolean` |
| `disabled` | `disabled` |             | `boolean` |
| `id`       | `id`       |             | `string`  |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
