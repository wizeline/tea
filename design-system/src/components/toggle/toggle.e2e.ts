import { newE2EPage } from '@stencil/core/testing';

describe('wz-toggle', () => {
  it('should render a toggle-component', async () => {
    const page = await newE2EPage();
    await page.setContent('<wz-toggle />');
    const toggle = await page.find('wz-toggle');
    expect(toggle.outerHTML).toMatchSnapshot();
  });
});
