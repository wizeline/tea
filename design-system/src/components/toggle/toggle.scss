wz-switch {
  --innerCircleSize: 20px;
  --switchWidth: 48px;
  --padding: 4px;
  --checkedPosition: calc(
    var(--switchWidth) - var(--innerCircleSize) - var(--padding)
  );

  input[type='checkbox'] {
    display: none;
  }

  label {
    content: '';
    display: flex;
    align-items: center;
    position: relative;
    width: var(--switchWidth);
    height: 28px;
    border-radius: 14px;
    background-color: var(--secondary-dark);
    cursor: pointer;
  }

  label:before {
    position: absolute;
    content: '';
    width: var(--innerCircleSize);
    height: var(--innerCircleSize);
    border-radius: 50%;
    background-color: var(--surface-base);
    @include transition(transform);
    transform: translateX(var(--padding));
  }

  input[type='checkbox']:checked + label {
    background-color: var(--primary-base);
    @include transition(background-color);
    &:before {
      transform: translateX(var(--checkedPosition));
    }
  }
}
