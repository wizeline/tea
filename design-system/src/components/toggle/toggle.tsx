import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'wz-toggle',
  styleUrl: 'toggle.scss',
  shadow: false,
})
export class Toggle {
  @Prop()
  id: string = 'wz-toggle';
  @Prop()
  checked: boolean = false;
  @Prop()
  disabled: boolean = false;

  render() {
    return (
      <div class="checkbox">
        <input
          checked={this.checked}
          disabled={this.disabled}
          id={this.id}
          type="checkbox"
        />
        <label htmlFor={this.id} />
      </div>
    );
  }
}
