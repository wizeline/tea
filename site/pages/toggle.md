### Toggle
```html
showSource: true
---
<wz-toggle checked />
```
