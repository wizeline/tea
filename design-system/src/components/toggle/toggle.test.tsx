import { Toggle } from './toggle';
import { TestWindow } from '@stencil/core/testing';

describe('wz-toggle', () => {
  it('renders value', async () => {
    const window = new TestWindow();
    const element = await window.load({
      components: [Toggle],
      html: `
        <wz-toggle />
      `,
    });

    expect(element).toMatchSnapshot();
  });
});
