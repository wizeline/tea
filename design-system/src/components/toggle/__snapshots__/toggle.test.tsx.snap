// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wz-toggle renders value 1`] = `
<wz-toggle
  class="hydrated"
>
  
  
      
  <div
    class="checkbox"
  >
    <input
      id=""
      type="checkbox"
    />
    <label
      for=""
    />
  </div>
</wz-toggle>
`;
